platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/model/machine/Command.java

```diff
@@ -37,4 +37,9 @@
      * Returns command type (i.e. 'maven')
      */
     String getType();
+
+    /**
+     * Returns command preview URL. (i.e. 'http://awesome.com:8080/context')
+     */
+    String getPreviewUrl();
```

Comment:
I don't like either. But according to issue we need to be able to store preview url for each command.